Tests/Functional/Eel/ElasticSearchQueryTest.php

```diff
@@ -57,26 +62,30 @@ class ElasticSearchQueryTest extends \TYPO3\Flow\Tests\FunctionalTestCase
     public function setUp()
     {
         parent::setUp();
+        $this->workspaceRepository = $this->objectManager->get('TYPO3\TYPO3CR\Domain\Repository\WorkspaceRepository');
+        $liveWorkspace = new \TYPO3\TYPO3CR\Domain\Model\Workspace("live");
+        $this->workspaceRepository->add($liveWorkspace);
+
         $this->nodeTypeManager = $this->objectManager->get('TYPO3\TYPO3CR\Domain\Service\NodeTypeManager');
         $this->contextFactory = $this->objectManager->get('TYPO3\TYPO3CR\Domain\Service\ContextFactoryInterface');
         $this->context = $this->contextFactory->create(array('workspaceName' => 'live', 'dimensions' => array('language' => array('de'))));
+
         $this->nodeDataRepository = $this->objectManager->get('TYPO3\TYPO3CR\Domain\Repository\NodeDataRepository');
         $this->queryBuilder = $this->objectManager->get('Flowpack\ElasticSearch\ContentRepositoryAdaptor\Eel\ElasticSearchQueryBuilder');
 
         $this->queryBuilder->log();
-        $this->createNodesForNodeSearchTest();
 
+        sleep(2); // we need to make sure that the index will be prefixed with an unique name
         $this->nodeIndexCommandController = $this->objectManager->get('Flowpack\ElasticSearch\ContentRepositoryAdaptor\Command\NodeIndexCommandController');
         $this->nodeIndexCommandController->buildCommand();
 
+        $this->createNodesForNodeSearchTest();
     }
 
-
     public function tearDown()
     {
         parent::tearDown();
         $this->inject($this->contextFactory, 'contextInstances', array());
-        $this->nodeIndexCommandController->cleanupCommand();
```

Comment:
By the way - we need a check for a running elasticsearch server during the setup and skip the tests if we do not find one....